traffic_ops/testing/api/v1/tenants_test.go

```diff
@@ -232,27 +232,31 @@ func UpdateTestTenantsActive(t *testing.T) {
 	}
 
 	// tenant3user with tenant3 has no access to ds3 with tenant3 when parent tenant2 is inactive
-	dses, _, err = tenant3Session.GetDeliveryServiceByXMLID("ds3")
+	dses, _, err = tenant3Session.GetDeliveryServiceByXMLIDNullable("ds3")
 	for _, ds := range dses {
-		t.Errorf("tenant3user got delivery service %+v with tenant3 but tenant3 parent tenant2 is inactive, expected: no ds", ds.XMLID)
+		if ds.XMLID != nil {
+			t.Errorf("tenant3user got delivery service %+v with tenant3 but tenant3 parent tenant2 is inactive, expected: no ds", *ds.XMLID)
+		}
 	}
 
 	setTenantActive(t, "tenant1", true)
 	setTenantActive(t, "tenant2", true)
 	setTenantActive(t, "tenant3", false)
 
 	// tenant3user with tenant3 has no access to ds3 with tenant3 when tenant3 is inactive
-	dses, _, err = tenant3Session.GetDeliveryServiceByXMLID("ds3")
+	dses, _, err = tenant3Session.GetDeliveryServiceByXMLIDNullable("ds3")
 	for _, ds := range dses {
-		t.Errorf("tenant3user got delivery service %+v with tenant3 but tenant3 is inactive, expected: no ds", ds.XMLID)
+		if ds.XMLID != nil {
+			t.Errorf("tenant3user got delivery service %+v with tenant3 but tenant3 is inactive, expected: no ds", *ds.XMLID)
```

Comment:
This is still an error if `ds.XMLID` is `nil` - we expected no Delivery Services returned. IMO, a better way to check this is by looking at the length of `dses` and maybe also not just ignoring the error. You don't have to make that change, but this does need to be an error whether or not `ds.XMLID` is `nil`.